aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -0,0 +1,183 @@
+/// This module supports functionality related to code management.
+module aptos_framework::code {
+    use std::string::String;
+    use std::error;
+    use std::signer;
+    use std::vector;
+
+    // ----------------------------------------------------------------------
+    // Code Publishing
+
+    /// The package registry at the given address.
+    struct PackageRegistry has key {
+        /// Packages installed at this address.
+        packages: vector<PackageMetadata>,
```

Comment:
High-level comment: How likely are we to get to deprecating the separate ModuleBundle upload transaction? If users upload modules through that transaction, it won't invoke the logic in this module and store the registry data right? If this is correct, it can lead it a weird state where some modules have these on chain registry data and some don't.

Reply:
All other code publishing should be turned off once this stabilizes.

Reply:
That's quite a big change but if you think we can get to it before AIT3, all is well 👍